ntoskrnl/se/sid.c

```diff
@@ -495,9 +525,29 @@ SeCaptureSidAndAttributesArray(
     ArraySize = AttributeCount * sizeof(SID_AND_ATTRIBUTES);
     RequiredLength = ALIGN_UP_BY(ArraySize, sizeof(ULONG));
 
-    /* Check for user mode data */
     if (PreviousMode != KernelMode)
     {
+        /* We're in user mode, set up the size for the temporary array */
+        TempArrayValidate = AttributeCount * sizeof(SID_VALIDATE_LENGTHS);
+        TempLengthValidate = ALIGN_UP_BY(TempArrayValidate, sizeof(ULONG));
+
+        /*
+         * Allocate a buffer for the array that
```

Comment:
you may be more flexible on the number of characters per line of your comments! (remember we usually use 80 chars per line)